src/cls/journal/cls_journal.cc

```diff
@@ -540,10 +540,12 @@ int journal_client_register(cls_method_context_t hctx, bufferlist *in,
                             bufferlist *out) {
   std::string id;
   bufferlist data;
+  std::set<cls::journal::Client> clients;
   try {
     bufferlist::iterator iter = in->begin();
     ::decode(id, iter);
     ::decode(data, iter);
+    ::decode(clients, iter);
```

Comment:
Ah! I see, `::decode(clients, iter)` could still return garbage. I'll make the required changes and resubmit (I had undone my earlier implementation using `journal_client_list`).